src/GafferUIBindings/PlugAdderBinding.cpp

```diff
@@ -89,13 +92,71 @@ struct PlugMenuSlotCaller
 
 };
 
+struct PlugAdderWrapper : public PlugAdder, IECorePython::Wrapper< PlugAdder >
+{
+
+	PlugAdderWrapper(PyObject *self, StandardNodeGadget::Edge edge ) : PlugAdder( edge ), IECorePython::Wrapper<PlugAdder>( self, this )
+	{
+	}
+
+	bool acceptsPlug( const Gaffer::Plug *connectionEndPoint ) const
+	{
+		IECorePython::ScopedGILLock gilLock;
+		override o = this->get_override( "acceptsPlug" );
+		if( o )
+		{
+			try
+			{
+				return o( Gaffer::PlugPtr( const_cast<Gaffer::Plug *>( connectionEndPoint ) ) );
+			}
+			catch ( error_already_set )
+			{
+				PyErr_Print();
+			}
+		}
+		else
+		{
+			// TODO: replace this with proper logging?
+			printf( "PlugAdder: acceptsPlug() python method not defined" );
+		}
+
+		return false;
+
+	}
+
+	virtual void addPlug( Gaffer::Plug *connectionEndPoint )
+	{
+		IECorePython::ScopedGILLock gilLock;
+		override o = this->get_override( "addPlug" );
+		if( o )
+		{
+			try
+			{
+				o( Gaffer::PlugPtr( const_cast<Gaffer::Plug *>( connectionEndPoint ) ) );
+			}
+			catch ( error_already_set )
+			{
+				PyErr_Print();
+			}
+		}
+		else
+		{
+			// TODO: replace this with proper logging?
+			printf( "PlugAdder: addPlug() python method not defined" );
+		}
+	}
+};
+
 } // namespace
 
 void GafferUIBindings::bindPlugAdder()
 {
-	scope s = GadgetClass<PlugAdder>()
+	scope s = GadgetClass<PlugAdder, PlugAdderWrapper>( "PlugAdder" )
+		.def( init<StandardNodeGadget::Edge>() )
 		.def( "plugMenuSignal", &PlugAdder::plugMenuSignal, return_value_policy<reference_existing_object>() )
 		.staticmethod( "plugMenuSignal" )
+		.def( "acceptsPlug", pure_virtual( &PlugAdderWrapper::acceptsPlug ) )
```

Comment:
We use it in cortex in some places [0] and the boost docs only say that "pure_virtual signals Boost.Python that the function f is a pure virtual function." [1]. I looked at the boost docs first, before looking at your wrappers, and thought `pure_virtual` would be needed. Maybe it's not?

[0] https://github.com/ImageEngine/cortex/blob/6fa38cea55d5e8bde8dec85e8aeQ707a757712bec/src/IECoreMaya/bindings/ImageViewportPostProcessBinding.cpp#L140
[1] http://www.boost.org/doc/libs/1_61_0/libs/python/doc/html/tutorial/tutorial/exposing.html